internal/provider/resource_bytes.go

```diff
@@ -0,0 +1,183 @@
+		ResultHex:    types.StringValue(hex.EncodeToString(bytes)),
+		Keepers:      plan.Keepers,
+	}
+
+	diags = resp.State.Set(ctx, u)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read does not need to perform any operations as the state in ReadResourceResponse is already populated.
+func (r *bytesResource) Read(context.Context, resource.ReadRequest, *resource.ReadResponse) {
+}
+
+// Update is intentionally left blank as all required and optional attributes force replacement of the resource
+// through the RequiresReplace AttributePlanModifier.
+func (r *bytesResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var model bytesModelV0
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+}
+
+// Delete does not need to explicitly call resp.State.RemoveResource() as this is automatically handled by the
+// [framework](https://github.com/hashicorp/terraform-plugin-framework/pull/301).
+func (r *bytesResource) Delete(context.Context, resource.DeleteRequest, *resource.DeleteResponse) {
+}
+
+func (r *bytesResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	bytes, err := base64.StdEncoding.DecodeString(req.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Import Random bytes Error",
+			"There was an error during the parsing of the base64 string.\n\n"+
+				diagnostics.RetryMsg+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return
+	}
+
+	var state bytesModelV0
+
+	state.ID = types.StringValue("none")
+	state.Length = types.Int64Value(int64(len(bytes)))
+	state.ResultBase64 = types.StringValue(req.ID)
+	state.ResultHex = types.StringValue(hex.EncodeToString(bytes))
+	state.Keepers = types.MapValueMust(types.StringType, nil)
+
+	diags := resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+type bytesModelV0 struct {
+	ID           types.String `tfsdk:"id"`
+	Length       types.Int64  `tfsdk:"length"`
+	Keepers      types.Map    `tfsdk:"keepers"`
+	ResultBase64 types.String `tfsdk:"result_base64"`
+	ResultHex    types.String `tfsdk:"result_hex"`
+}
+
+func bytesSchemaV0() schema.Schema {
+	return schema.Schema{
+		Version: 0,
+		Description: "The resource `random_bytes` generates random bytes that are intended to be " +
+			"used as secret or keys.",
+		Attributes: map[string]schema.Attribute{
+			"keepers": schema.MapAttribute{
+				Description: "Arbitrary map of values that, when changed, will trigger recreation of " +
+					"resource. See [the main provider documentation](../index.html) for more information.",
+				ElementType: types.StringType,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Map{
+					mapplanmodifiers.RequiresReplaceIfValuesNotNull(),
+				},
```

Comment:
Drive-by comment: This new resource could use the regular `mapplanmodifier.RequiresReplace()` as there shouldn't be any prior state storage from a terraform-plugin-sdk based resource which would've collapsed a map of null values into a null map (the intended usage of the custom plan modifier was to prevent replacing existing resources stored with the prior SDK's invalid state). It may be worth an additional code comment here on the design decision either which way, whether to keep consistency with the other resources or not (so all `keepers` changes including adding `null` values replace the resource).

Reply:
@bendbennett should I change this or keep it that way ?

Reply:
@Socolin just for the sake of consistency with the other resources, I think I'd be inclined to use `mapplanmodifier.RequiresReplace()` and add a code comment as @bflad suggests.

Reply:
What the comment should say ?  something like ?:

`mapplanmodifier.RequiresReplace()` could be used, but for consistency with other resources we use  `mapplanmodifiers.RequiresReplaceIfValuesNotNull() ` 

or do you have any other suggestion ? @bendbennett

Reply:
@Socolin it might be useful to refer to the fact that _random_bytes_ was not available as a a resource on the _random_ provider until after the migration from SDKv2 to the Framework. 

Perhaps something like:
> mapplanmodifiers.RequiresReplaceIfValuesNotNull() has been used for consistency with other resources but mapplanmodifier.RequiresReplace() could have been used as there shouldn't be any prior state storage from a terraform-plugin-sdk based resource which would've collapsed a map of null values into a null map.

Reply:
Ok, thanks.